app/player/knight.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, name: str, power: int, hp: int, armour: list,
+                 weapon: dict, potion: dict) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+
+        for item in armour:
+            self.protection += item["protection"]
+
+        self.power += weapon["power"]
+
+        if potion is not None:
+            if "power" in potion["effect"]:
+                self.power += potion["effect"]["power"]
+
+            if "protection" in potion["effect"]:
+                self.protection += potion["effect"]["protection"]
+
+            if "hp" in potion["effect"]:
+                self.hp += potion["effect"]["hp"]

```

Comment:
Use loop, `settatr()` can help you

![image](https://user-images.githubusercontent.com/104218550/215174781-ef910885-508a-471e-9164-ba6a19580dab.png)